es5-persistence/src/main/java/com/netflix/conductor/elasticsearch/es5/EmbeddedElasticSearchV5.java

```diff
@@ -0,0 +1,123 @@
+
+    private Node instance;
+    private Client client;
+    private File dataDir;
+
+    private class PluginConfigurableNode extends Node {
+        public PluginConfigurableNode(Settings preparedSettings, Collection<Class<? extends Plugin>> classpathPlugins) {
+            super(InternalSettingsPreparer.prepareEnvironment(preparedSettings, null), classpathPlugins);
+        }
+    }
+
+    @Override
+    public void start() throws Exception {
+        start(DEFAULT_CLUSTER_NAME, DEFAULT_HOST, DEFAULT_PORT, true);
+    }
+
+    public synchronized void start(String clusterName, String host, int port, boolean enableTransportClient) throws Exception {
+
+        if (instance != null && !instance.isClosed()) {
+            logger.info("Elastic Search is already running on port {}", getPort());
+            return;
+        }
+
+        final Settings settings = getSettings(clusterName, host, port, enableTransportClient);
+        dataDir = setupDataDir(settings.get(ES_PATH_DATA));
+
+        logger.info("Starting ElasticSearch for cluster {} ", settings.get("cluster.name"));
+        instance = new PluginConfigurableNode(settings, singletonList(Netty4Plugin.class));
+        instance.start();
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                try {
+                    instance.close();
+                } catch (IOException e) {
+                    logger.error("Error closing ElasticSearch");
+                }
+            }
+        });
+        logger.info("ElasticSearch cluster {} started in local mode on port {}", instance.settings().get("cluster.name"), getPort());
+        client = instance.client();
+    }
+
+    private Settings getSettings(String clusterName, String host, int port, boolean enableTransportClient) throws IOException {
+        dataDir = Files.createTempDirectory(clusterName + "_" + System.currentTimeMillis() + "data").toFile();
+        File homeDir = Files.createTempDirectory(clusterName + "_" + System.currentTimeMillis() + "-home").toFile();
+        Settings.Builder settingsBuilder = Settings.builder()
+                .put("cluster.name", clusterName)
+                .put("http.host", host)
+                .put("http.port", port)
+                .put(ES_PATH_DATA, dataDir.getAbsolutePath())
+                .put(ES_PATH_HOME, homeDir.getAbsolutePath())
+                .put("http.enabled", true)
+                .put("script.inline", true)
+                .put("script.stored", true)
+                .put("node.data", true)
+                .put("http.enabled", true)
+                .put("http.type", "netty4")
+                .put("transport.type", "netty4");
+
+        return settingsBuilder.build();
+    }
+
+    public Client getClient() {
+        if (instance == null || instance.isClosed()) {
+            logger.error("Embedded ElasticSearch is not Initialized and started, please call start() method first");
```

Comment:
This really ought to be a `throw`, right? Returning `null` here is only going to make debugging the issue harder, as the server will crash with a `NullPointerException` further down the line.

Reply:
As far as I can tell this code isn't even used, so maybe it would be better to just delete it?